armi/reactor/converters/axialExpansionChanger/axialExpansionChanger.py

```diff
@@ -413,17 +483,294 @@ def manageCoreMesh(self, r):
             for old, new in zip(oldMesh, r.core.p.axialMesh):
                 runLog.extra(f"{old:.6e}\t{new:.6e}")
 
+    def _checkComponentHeight(self, c):
+        if c.zbottom > c.ztop:
+            msg = f"""
+            {c} has a negative height! This is unphysical.
+                Assembly: {self.linked.a}
+                    Block: {c.parent}
+                Component: {c}
 
-def _checkBlockHeight(b):
-    """
-    Do some basic block height validation.
+                Component Height = {c.ztop} - {c.zbottom} = {c.height}.
+            """
+            raise ArithmeticError(dedent(msg))
 
-    Notes
-    -----
-    3cm is a presumptive lower threshold for DIF3D
+    def _checkBlockHeight(self, b):
+        """
+        Do some basic block height validation.
+
+        Notes
+        -----
+        3cm is a presumptive lower threshold for DIF3D
+        """
+        if b.getHeight() < 3.0:
```

Comment:
Magic Number Alert.

Can we move this `3.0` to a class-level or global constant in this file?

Perhaps it could be named something like `DIF3D_MIN_BLOCK_HEIGHT`.